### PR TITLE
fix replacing mixed data types

### PR DESCRIPTION
**Pull request by Gazler**
Previously, attempting to replace a string with a map when performing
the recursive merge would result in a JavaScript error:

Uncaught TypeError: Cannot assign to read only property '0'
of string '<button type="submit">Press Me</button>'

This could be triggered with the following LiveView.

```elixir
defmodule LiveViewBugWeb.CounterLive do
  use Phoenix.LiveView

  def render(assigns) do
    ~L"""
    <form phx-submit="generate">
      <%= if @value do %>
        <%= if true do %>
          <%= @value %>
        <% end %>
      <% else %>
        <%= Phoenix.HTML.Form.submit "Press Me" %>
      <% end %>
    </form>
    """
  end

  def mount(_session, socket) do
    {:ok, assign(socket, :value, nil)}
  end

  def handle_event("generate", params, socket) do
    {:noreply, assign(socket, :value, 1)}
  end
end
```

To fix this, the type of the `targetVal` is also checked when performing
a recursive merge. If the `targetVal` is not an object, then it will
simply be replaced by the new `val.

### CODE REVIEW

File: assets/js/phoenix_live_view.js [hunk 1]
```diff
@@ -177,7 +177,7 @@ let recursiveMerge = (target, source) => {
   for(let key in source){
     let val = source[key]
     let targetVal = target[key]
-    if(isObject(val) && targetVal){
+    if(isObject(val) && isObject(targetVal)){
       if(isObject(targetVal) && targetVal.dynamics && !val.dynamics){ delete targetVal.dynamics}
```
Comment (josevalim) [inline assets/js/phoenix_live_view.js]:
```suggestion
      if(targetVal.dynamics && !val.dynamics){ delete targetVal.dynamics}
```

**Comment by mcade** [comment]
Ran into this issue today. Any reason in particular this commit hasn't been merged yet or is it only a matter of time? I just tried using the gr-replace-string branch in mix.exs and it fixed the js error I was getting for `Cannot assign to read only property ...`

**Comment by oskarkook** [comment]
Could we get this merged? It fixes #266

**Comment by chrismccord** [comment]
❤️❤️❤️🐥🔥

**Pull request merged**